### ISSUE TITLE
Bug in custom Validation 

### ISSUE THREAD

**Issue by abhiagwl4262** [ultralytics/ultralytics#17659]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Val

### Bug

The inputs in `BaseValidator` class `__call__` function are `model and trainer`
https://github.com/ultralytics/ultralytics/blob/652e05e833087652053346946827034a4a038025/ultralytics/engine/validator.py#L107

while you are using `args and _callbacks` here https://github.com/ultralytics/ultralytics/blob/652e05e833087652053346946827034a4a038025/ultralytics/engine/model.py#L634

### Environment

OS                  Linux-5.15.0-1033-aws-x86_64-with-glibc2.31
Environment         Docker
Python              3.10.13
Install             pip
RAM                 15.34 GB
Disk                228.1/248.1 GB
CPU                 Intel Xeon Platinum 8259CL 2.50GHz
CPU count           4
GPU                 Tesla T4, 15102MiB
GPU count           1
CUDA                12.1

numpy               ✅ 1.23.5>=1.23.0
matplotlib          ✅ 3.8.2>=3.3.0
opencv-python       ✅ 4.8.1.78>=4.6.0
pillow              ✅ 10.0.1>=7.1.2
pyyaml              ✅ 6.0>=5.3.1
requests            ✅ 2.31.0>=2.23.0
scipy               ✅ 1.11.4>=1.4.1
torch               ✅ 2.1.0>=1.8.0
torchvision         ✅ 0.16.0>=0.9.0
tqdm                ✅ 4.65.0>=4.64.0
psutil              ✅ 5.9.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.1.4>=1.1.4
seaborn             ✅ 0.13.0>=0.11.0
ultralytics-thop    ✅ 2.0.9>=2.0.0
torch               ✅ 2.1.0!=2.4.0,>=1.8.0; sys_platform == "win32"

### Minimal Reproducible Example

from ultralytics import YOLO
from ultralytics.models.yolo.segment.val import SegmentationValidator
from ultralytics.engine.results import Results
# from ultralytics.utils.ops import Profile
# from pathlib import Path

def your_custom_postprocess_function(pred):
    return pred

class CustomSegmentationValidator(SegmentationValidator):
    def __init__(self, dataloader=None, save_dir=None, pbar=None, args=None, _callbacks=None):
        # Initialize with default values if not provided
        if args is None:
            args = self.get_default_args()
        super().__init__(dataloader=dataloader, save_dir=save_dir, pbar=pbar, args=args, _callbacks=_callbacks)
    
    @staticmethod
    def get_default_args():
        """Define default arguments for the validator."""
        return {
            'data': None,  # path to data file, i.e. coco.yaml
            'imgsz': 640,  # image size as integer or w,h
            'batch': 16,  # number of images per batch
            'device': '',  # device to run on, i.e. cuda device=0/1/2/3 or device=cpu
            'workers': 8,  # number of worker threads for data loading
            'verbose': True,  # whether to print verbose output
            'project': 'runs/val',  # project name
            'name': 'exp',  # experiment name
            'exist_ok': False,  # whether to overwrite existing experiment
            'half': False,  # whether to use half precision (FP16)
            'dnn': False,  # whether to use OpenCV DNN for ONNX inference
            'plots': True,  # whether to generate plots during validation
            'rect': False,  # rectangular training
            'split': 'val',  # dataset split to use for validation (train/val/test)
        }
    def __call__(self, args=None, model=None, _callbacks=None):
        super().__call__(model=model)

    def postprocess(self, preds):
        """
        Apply custom post-processing to the raw predictions.
        Args:
            preds (list): List of predictions from the model
        Returns:
            list: Post-processed predictions
        """
        parent_processed_preds = super().postprocess(preds)
        custom_processed_preds = []
        for pred in parent_processed_preds:
            processed_pred = your_custom_postprocess_function(pred)
            custom_processed_preds.append(processed_pred)

        results = []
        for pred in custom_processed_preds:
            result = Results(
                boxes=pred.boxes,
                masks=pred.masks,
                probs=pred.probs if hasattr(pred, 'probs') else None
            )
            results.append(result)

        return results

# Example usage:
def run_validation():
    # Load your model
    model = YOLO(<model_path>)
    
    # Setup validation arguments
    val_args = {
        'data': 'data_v4/data.yaml',  # Path to your data configuration file
        'imgsz': 640,                      # Image size
        'batch': 16,                       # Batch size
        'device': '0',                     # GPU device (use '' for CPU)
        'workers': 8,                      # Number of worker threads
        'project': 'runs/val',             # Project directory
        'name': 'custom_validation',       # Run name
        'exist_ok': True,                  # Overwrite existing results
    }
    
    # Create custom validator instance
    # custom_validator = CustomSegmentationValidator(args=val_args)
    custom_validator = CustomSegmentationValidator()
    
    breakpoint()
    # Run validation
    metrics = model.val(validator=custom_validator)
    
    return metrics

if __name__ == "__main__":
    metrics = run_validation()

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @abhiagwl4262, thank you for bringing this to our attention! 🚀 This is an automated response to assist you promptly. An Ultralytics engineer will take a closer look at your issue soon.

We recommend checking out our [documentation](https://docs.ultralytics.com) for in-depth guidance on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage. If you haven't already, you might find solutions or workarounds to your issue there.

🛠️ For this 🐛 Bug Report, we appreciate the minimal reproducible example (MRE) you provided. This helps us greatly in diagnosing the issue.

In the meantime, please ensure you have the latest version of Ultralytics installed. Upgrade using the command below to verify your issue isn't already resolved:

```bash
pip install -U ultralytics
```

To experiment with YOLO in verified environments, you can try:

- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: Refer to our [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **AWS**: Consult our [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: See our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/)

For real-time support, feel free to join the conversation on [Discord](https://ultralytics.com/discord) 🎧. Alternatively, engage with the community on our [Discourse](https://community.ultralytics.com) or explore topics on our [Subreddit](https://reddit.com/r/ultralytics).

![Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg)

If this badge is green, our tests are passing, indicating that the framework should be operational in the latest build. 

Thank you for your patience and contributions to improving Ultralytics!
